src/test/java/org/folio/rest/api/ItemDamagedStatusAPITest.java

```diff
@@ -0,0 +1,247 @@
+      .withSource("local");
+
+    client.post(itemDamagedStatusesUrl(EMPTY), status, TEST_TENANT)
+      .whenComplete((response, throwable) -> {
+        assertThat(response, notNullValue());
+        assertThat(response.getStatusCode(), is(HttpStatus.HTTP_CREATED.toInt()));
+      })
+      .thenCompose(response -> {
+        String id = response.getJson().getString("id");
+        return client.get(itemDamagedStatusesUrl("/" + id), TEST_TENANT);
+      })
+      .whenComplete((response, throwable) -> {
+        assertThat(response, notNullValue());
+        assertThat(response.getStatusCode(), is(HttpStatus.HTTP_OK.toInt()));
+        JsonObject statusJson = response.getJson();
+        assertThat(statusJson.getString("id"), notNullValue());
+        assertThat(statusJson.getString("name"), is(status.getName()));
+      });
+  }
+
+  @Test
+  public void canGetAllItemDamagedStatuses() {
+    ItemDamageStatus status = new ItemDamageStatus()
+      .withName("test item damaged status name")
+      .withSource("local");
+
+    client.post(itemDamagedStatusesUrl(EMPTY), status, TEST_TENANT)
+      .whenComplete((response, throwable) -> {
+        assertThat(response, notNullValue());
+        assertThat(response.getStatusCode(), is(HttpStatus.HTTP_CREATED.toInt()));
+      })
+      .thenCompose(response -> {
+        status.setName("second test item damaged status name");
+        return client.post(itemDamagedStatusesUrl(EMPTY), status, TEST_TENANT);
+      })
+      .whenComplete((response, throwable) -> {
+        assertThat(response, notNullValue());
+        assertThat(response.getStatusCode(), is(HttpStatus.HTTP_CREATED.toInt()));
+      })
+      .thenCompose(it -> client.get(itemDamagedStatusesUrl(EMPTY), TEST_TENANT))
+      .whenComplete((response, throwable) -> {
+        int size = response.getJson().getJsonArray("itemDamageStatuses").size();
+        assertThat(size, is(2));
+      });
+  }
+
+  @Test
+  public void shouldFindItemDamagedStatusesByName() {
+    ItemDamageStatus status = new ItemDamageStatus()
+      .withName("test item damaged status name")
+      .withSource("local");
+
+    client.post(itemDamagedStatusesUrl(EMPTY), status, TEST_TENANT)
+      .whenComplete((response, throwable) -> {
+        assertThat(response, notNullValue());
+        assertThat(response.getStatusCode(), is(HttpStatus.HTTP_CREATED.toInt()));
+      })
+      .thenCompose(response -> {
+        status.setName("itemDamagedStatusName");
+        return client.post(itemDamagedStatusesUrl(EMPTY), status, TEST_TENANT);
+      })
+      .whenComplete((response, throwable) -> {
+        assertThat(response, notNullValue());
+        assertThat(response.getStatusCode(), is(HttpStatus.HTTP_CREATED.toInt()));
+      })
+      .thenCompose(it -> {
+        URL url = itemDamagedStatusesUrl("?query=name=itemDamagedStatusName");
+        return client.get(url, TEST_TENANT);
+      })
+      .whenComplete((response, throwable) -> {
```

Comment:
Is this chaining style common in the tests of other projects?

Personally I find it harder to read than sequential blocking of asynchronous requests.

Reply:
I don't think that it's common style. Shall rewrite it ?

Reply:
I think it would be preferable for it to closer resemble the existing tests in the module, especially as it seems unlikely they will all be converted to use this style.

Reply:
if you say so. The third time I will rewrite these test. I think it will be the last one

Reply:
I'm not trying to force you to rewrite things multiple times. I'm trying to share my thoughts, ask questions and understand decisions.

What I don't understand with these tests is why you chose to undertake a different style to the existing tests? Do you find this style easier to read, you've used it elsewhere?

Reply:
the first call returns a future as you asked me to do for the HttpClient to avoid callbacks.
A future has appropriate methods to handle it
The main reason is to avid async method call on TestContext object.
All productive ready code has chain-style approach. Why don't it use in tests? 
I find it well-structured, easy read, async styled approach.

Reply:
> the first call returns a future as you asked me to do for the HttpClient to avoid callbacks.

Sure, and I'm glad for that change. In retrospect, I would've preferred to use completable futures from the beginning (I was trying to stick to FOLIO's decision to use vert.x).

> A future has appropriate methods to handle it
> The main reason is to avid async method call on TestContext object.

The TestContext that is provided by vert.x testing tools?

If so, I agree with you, I find the vert.x testing tools to be especially difficult to follow.

> All productive ready code has chain-style approach. Why don't it use in tests?
> I find it well-structured, easy read, async styled approach.

I'll admit I have different beliefs about how test code and production code read.

It may well be that I've given an unfair comparison to the other code in mod-inventory-storage.

I want tests to always produce an output. Asynchronously chained code can easily not do so. For example, if the future in that chain does not complete, what happens during the test?

I would hope that most of the production code that uses this style (there is lots in FOLIO that does not) uses method references or single line lambda statements for the chained steps.

I find the multi-line nested lambda statements difficult to read, with the content obscured by the noise of the callback-esque indentation and brackets.

Does any of that make sense?

Reply:
Would it be useful to get feedback from other folks, given that I'm the only one who has interacted with this pull request?

Reply:
> I want tests to always produce an output. Asynchronously chained code can easily not do so. For example, if the future in that chain does not complete, what happens during the test?

This is a good point. To prevent this we can use it at the end of a chain.
```future.get(5, TimeUnit.SECONDS);```

Reply:
Please give that a try, and check that the test finishes with a timeout if the implementation does not respond.